docs/doxygen/pages.dox

```diff
@@ -0,0 +1,98 @@
+/**
+@mainpage Overview
+@anchor mqtt
+@brief Thread safe MQTT 3.1.1 client
+
+coreMQTT is an MIT licensed open source C MQTT client library for microcontroller and small microprocessor based IoT devices. It’s design is intentionally simple to ensure it has no dependency on any other library or operating system, and to better enable static analysis including memory safety proofs. That simplicity and lack of operating system dependency (coreMQTT does not require multithreading at all) means coreMQTT does not build thread safety directly into its implementation. Instead thread safety must be provided by higher level software. This web page demonstrates a coreMQTT extension that provides that higher level functionality in the form of an MQTT agent (or MQTT daemon). While the implementation demonstrated here is currently specific to FreeRTOS, there are not many dependencies on FreeRTOS, meaning the implementation can easily be adapted for use with other operating systems.
```

Comment:
With initial description about just the coreMQTT library, the emphasis on coreMQTT Agent could be missing for a new reader. Would it be possible to highlight about MQTT Agent and separately have a section about why a newer library on top of coreMQTT library has been created? Basically, something of this structure?

**Overview**

_Description about MQTT Agent which says that it is based on the coreMQTT library and provides thread-safe implementation of an MQTT client on the FreeRTOS operating system_

**Why is there a higher level library based besides coreMQTT?**

_Brief description about coreMQTT, its benefits and its limitations_

Reply:
I put them in separate sections